Added support for dot indexing and multiple index levels using recursion.

DionTimmermann opened the pull request:
These changes also make the variable editor behave properly for LinProp variables.

These changes have not yet been tested intensively, i.e. I have not yet created a test script. But simple code like this now works:
```
clear

a = LinProp(5);
a.Value

b = [LinProp(5) LinProp(6) LinProp(7)];
b.Value
b(2:3).Value
```

For now, I have only editied LinProp.m. Once there are no remaining issues, I will copy the changes to DistProp and MCProp.

The most important change was to overload `numArgumentsFromSubscript`. As that function's use is really confusing for me, I added relatively detailed explanation. 

Further minor issues/ToDos:
* I also set NetObject to hidden, so it does not show in the variable editor.
* I am unsure if isArray should really be visible. It is rather an internal property of the variable, than one that is important for normal users.
* It now might also be possible to allow set access for Value and StdUnc, although I am not sure, if that is really useful...

wollmich commented:
Dot indexing was working with [V2.4.7](https://github.com/wollmich/metas-unclib-matlab-wrapper/releases/tag/V2.4.7) for the following properties NetObject, Value, StdUnc, IsComplex and IsArray.

wollmich commented:
I would prefer that you don't hide any properties. First fix the dot indexing. There were some reason in the past to access for example the NetObject property.

DionTimmermann commented:
Yes, had `numArgumentsFromSubscript` been defined, dot indexing would have also worked for matricies in 2.4.7. When I reworked subsref, I did not notice that dot indexing worked for scalars and thus removed it completely. I added it back in in this branch. It now works the same as in 2.4.7, i.e. `B=subsref(A, S)` directly calls `B=A.(S.subs)`. Which properties are visible is determined by the getAccess attribute.

Multilevel indexing is added in this branch, as I initially thought the two issues would be inseperable. (Without `numArgumentsFromSubscript` the dot index for a matrix always resulted in multilevel indexing.)

If it helps, I can also close this pull request and make two idenpendent ones.

wollmich commented:
> If it helps, I can also close this pull request and make two independent ones.
I think there is no need to sperate it into two pull requests.

I'll take a look at it tomorrow.

wollmich commented:
> For now, I have only editied LinProp.m. Once there are no remaining issues, I will copy the changes to DistProp and MCProp.

I've tested the changes you did for LinProp.m and they passed. So you can copy the changes to DistProp.m and MCProp.m.

wollmich commented on @LinProp/LinProp.m:
```diff
@@ -684,6 +684,20 @@ function display(obj)
+            % function would retrun 1. 
+            %
+            % This class does not support brace indexing. When using dot
+            % indexing on a LinProp matrix, e.g. a = Linprop(1:3); a.Value, 
```
Typo: `a = LinProp(1:3);`
